cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,92 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
     
+    cy.visit('/');
+    cy.get('h5')
+      .should('include.text', 'Student Registration Form');
+    cy.task('generateUser')
+      .then(generateUser => {
+        user = generateUser;
+      });
   });
 
-  it('', () => {
+  it('should provide an ability to register', () => {
+
+    cy.findByPlaceholder('First Name')
+      .type(user.firstName);
+
+    cy.findByPlaceholder('Last Name')
+      .type(user.lastName);
     
+    cy.get('#userEmail')
+      .type(user.email);
+
+    cy.get(`[value = ${user.gender}]`)
+      .click({ force: true });
+
+    cy.contains('.custom-control-label', user.gender)
```

Comment:
You can remove this redundant line, as the gender radio button is already clicked in the line above.